tests/rm_files.rs

```diff
@@ -0,0 +1,234 @@
+            Component::CurDir => ".",
+            Component::ParentDir => "..",
+            Component::Normal(comp) => comp.to_str().unwrap(),
+        })
+        .map(|comp| utf8_percent_encode(comp, percent_encode_sets::COMPONENT).to_string())
+        .collect::<Vec<_>>()
+        .join("/")
+}
+
+/// Construct a path for a deletion POST request without any further encoding.
+///
+/// This should be kept consistent with implementation.
+fn make_del_path(unencoded_path: impl AsRef<Path>) -> String {
+    format!("rm?path=/{}", make_get_path(unencoded_path))
+}
+
+fn assert_rm_ok(base_url: Url, unencoded_paths: &[impl AsRef<Path>]) -> anyhow::Result<()> {
+    let client = Client::new();
+
+    for file_path in unencoded_paths.iter().map(AsRef::as_ref) {
+        // encode
+        let get_url = base_url.join(&make_get_path(file_path))?;
+        let del_url = base_url.join(&make_del_path(file_path))?;
+        println!("===== {file_path:?} =====");
+        println!("{get_url}, {del_url}");
+
+        // check path exists
+        let _get_res = client.get(get_url.clone()).send()?.error_for_status()?;
+
+        // delete
+        let _del_res = client.post(del_url).send()?.error_for_status()?;
+
+        // check path is gone
+        let get_res = client.get(get_url).send()?;
+        if get_res.status() != StatusCode::NOT_FOUND {
+            bail!("Unexpected status code: {}", get_res.status());
+        }
+    }
+
+    Ok(())
+}
+
+/// The `check_paths_exist` parameter allows skipping this check before and after
+/// the deletion attempt in case these paths should be inaccessible via GET.
+fn assert_rm_err(
+    base_url: Url,
+    unencoded_paths: &[impl AsRef<Path>],
+    check_paths_exist: bool,
+) -> anyhow::Result<()> {
+    let client = Client::new();
+
+    for file_path in unencoded_paths.iter().map(AsRef::as_ref) {
+        // encode
+        let get_url = base_url.join(&make_get_path(file_path))?;
+        let del_url = base_url.join(&make_del_path(file_path))?;
+        println!("===== {file_path:?} =====");
+        println!("{get_url}, {del_url}");
+
+        // check path exists
+        if check_paths_exist {
+            let _get_res = client.get(get_url.clone()).send()?.error_for_status()?;
+        }
+
+        // delete
+        let del_res = client.post(del_url).send()?;
+        if !del_res.status().is_client_error() {
+            bail!("Unexpected status code: {}", del_res.status());
+        }
+
+        // check path still exists
+        if check_paths_exist {
+            let _get_res = client.get(get_url).send()?.error_for_status()?;
+        }
+    }
+
+    Ok(())
+}
+
+#[rstest]
+fn rm_disabled_by_default(server: TestServer) -> anyhow::Result<()> {
+    let paths = [FILES, DIRECTORIES]
+        .concat()
+        .into_iter()
+        .map(Path::new)
+        .chain(iter::once(DEEPLY_NESTED_FILE.as_ref()))
+        .collect::<Vec<_>>();
+    assert_rm_err(server.url(), &paths, true)
```

Comment:
Hm, for all of these days I wonder whether it wouldn't be nicer and more obvious to use rstest more in the way that it was intended and have test-cases generated? Currently all paths are tested in the same test which is not really the intention of rstest. Maybe take a look at how we do things in `upload_files.rs` and try to follow that?